extra/Addons/Macros/SelectionStartEnd.lua

```diff
@@ -0,0 +1,96 @@
+﻿-- Selection Start/End with hot-keys Ctrl+[ / Ctrl+] or via Macros Menu
+Color = far.AdvControl(far.Flags.ACTL_GETCOLOR,far.Colors.COL_HELPTOPIC)
+Selection = {}
+
+Event {
+  group="EditorEvent";
+  action=function(id,Event,param)
+    local info = editor.GetInfo()
+    if Event==far.Flags.EE_READ then
+      Selection[info.EditorID] = {
+        BlockType=far.Flags.BTYPE_STREAM,
+       	StartLine=1,
+        StartPos=1,
+       	EndLine=1,
+        EndPos=0
+      }
+    end
+    if Event==far.Flags.EE_CLOSE then 
+      Selection[info.EditorID]=nil
+    end
+  end
+}
+
+local function Hint(str)
+  editor.Redraw()
+  far.Text(2,0,Color,str)
+  far.Text()
+  win.Sleep(100)
+end
+
+local function changeBlock(oldSelection,btype)
+  local startpos = btype==far.Flags.BTYPE_STREAM and oldSelection.StartPos
+                or btype==far.Flags.BTYPE_COLUMN and editor.RealToTab(nil,oldSelection.StartLine,oldSelection.StartPos)
+  local endpos   = btype==far.Flags.BTYPE_STREAM and oldSelection.EndPos
+                or btype==far.Flags.BTYPE_COLUMN and editor.RealToTab(nil,oldSelection.EndLine,oldSelection.EndPos+1)-1
+  local newSelection = {
+    BlockType=btype,
+    BlockStartPos=startpos,
+    BlockWidth=endpos-startpos+1,
+    BlockStartLine=oldSelection.StartLine,
+    BlockHeight=oldSelection.EndLine-oldSelection.StartLine+1,
+  }
+
+  local visible = newSelection.BlockWidth~=0 or newSelection.BlockHeight>1
+  visible = editor.Select(nil,newSelection) and visible
+  if not visible then editor.Select(nil,far.Flags.BTYPE_NONE) end
+  return visible
+end
+
+local function startSelection()
+  local info = editor.GetInfo()
+  Selection[info.EditorID].StartLine = info.CurLine
+  Selection[info.EditorID].StartPos = info.CurPos
+  changeBlock(Selection[info.EditorID],far.Flags.BTYPE_STREAM)
+  Hint"Selection start"
+end
+
+local function endSelection()
+  local info = editor.GetInfo()
+  Selection[info.EditorID].EndLine = info.CurLine
+  Selection[info.EditorID].EndPos = info.CurPos - 1
+  changeBlock(Selection[info.EditorID],far.Flags.BTYPE_STREAM)
+  Hint"Selection end"
+end
+
+Macro {
+  description="^Ctrl[ Selection start";
+  area="Editor";
+  key="Ctrl[";
+  action=startSelection;
+}
+
+Macro {
+  description="^Ctrl] Selection end";
+  area="Editor";
+  key="Ctrl]";
+  action=endSelection;
+}
+
+MenuItem {
+  description = "Selection &[ start";
+  menu   = "Plugins";
+  area   = "Editor";
+  guid   = "A435D561-AD64-4DD1-8C61-28CB90358817";
+  text   = "Selection &[ start";
+  action=startSelection;
+}
+
+MenuItem {
+  description = "Selection &] end";
+  menu   = "Plugins";
+  area   = "Editor";
+  guid   = "A435D561-AD64-4DD1-8C61-28CB90358818";
+  text   = "Selection &] end";
+  action=startSelection;
```

Comment:
`=` alignment here is not consistent.
(I suppose we need no alignment, as no other macro has it)